File: finance-app-backend/src/main/java/com/victor/financeapp/backend/application/dto/CreditCardDTO.java
package com.victor.financeapp.backend.application.dto;

import lombok.Builder;

import java.math.BigDecimal;
import java.util.List;

@Builder
public record CreditCardDTO(
        Long id,
        Long userId,
        String title,
        String description,
        String number,
        String color,
        Integer invoiceClosingDay,
        BigDecimal totalInvoiceAmount,
        List<InvoiceDTO> invoices
) {
}

File: finance-app-android/app/src/main/java/com/viictrp/financeapp/application/dto/CreditCardDTO.kt
package com.viictrp.financeapp.application.dto

import java.math.BigDecimal

data class CreditCardDTO(
    val id: Long,
    val title: String,
    val description: String,
    val color: String,
    val invoiceClosingDay: Int,
    val totalInvoiceAmount: BigDecimal,
    val invoices: List<InvoiceDTO> = emptyList()
)
